api/envoy/extensions/filters/http/bandwidth_limit/v3alpha/bandwidth_limit.proto

```diff
@@ -0,0 +1,70 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.bandwidth_limit.v3alpha;
+
+import "envoy/config/core/v3/base.proto";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.bandwidth_limit.v3alpha";
+option java_outer_classname = "BandwidthLimitProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Bandwidth limit]
+// Bandwidth limit :ref:`configuration overview <config_http_filters_bandwidth_limit>`.
+// [#extension: envoy.filters.http.bandwidth_limit]
+
+// [#next-free-field: 6]
+message BandwidthLimit {
+  // Defines the mode for the bandwidth limit filter.
+  // Values represent bitmask.
+  enum EnableMode {
+    // Filter is disabled.
+    DISABLED = 0;
+
+    // Filter enabled only for incoming traffic.
+    DECODE = 1;
+
+    // Filter enabled only for outgoing traffic.
+    ENCODE = 2;
+
+    // Filter enabled for both incoming and outgoing traffic.
+    DECODE_AND_ENCODE = 3;
```

Comment:
I chose it since Envoy can be used as ingress or egress proxy where request/response could mean different things. Open to changing it if you think this doesn't make sense.

Reply:
Renamed on further thoughts and after seeing request/response usage in current APIs.